tests/MongoDB.Driver.Core.Tests/Core/Configuration/ClusterBuilderTests.cs

```diff
@@ -59,10 +63,44 @@ public void CreateServerMonitorFactory_should_return_expected_result(int connect
 
             var serverSettings = result._settings();
         }
+
+        [Fact]
+        public void RegisterStreamFactory_should_complement_previous_RegisterStreamFactory_call()
+        {
+            string testValue = "TestValue";
+
+            var subject = new ClusterBuilder();
+            subject.RegisterStreamFactory(factory => new StreamFactoryTest { TestValue = testValue });
+            subject.RegisterStreamFactory(factory =>
+            {
+                var testFactory = (StreamFactoryTest)factory;
+                testFactory.TestValue.Should().Be(testValue);
+                return factory;
+            });
+            var streamFactoryWrapper = subject.GetStreamFactoryWrapperFunc();
+            streamFactoryWrapper(null);
+        }
+    }
+
+    public class StreamFactoryTest : IStreamFactory
+    {
+        public Stream CreateStream(EndPoint endPoint, CancellationToken cancellationToken)
+        {
+            throw new NotImplementedException();
+        }
+
+        public Task<Stream> CreateStreamAsync(EndPoint endPoint, CancellationToken cancellationToken)
+        {
+            throw new NotImplementedException();
+        }
+
+        public string TestValue { get; set; }
     }
 
     public static class ClusterBuilderReflector
     {
         internal static IServerMonitorFactory CreateServerMonitorFactory(this ClusterBuilder obj) => (IServerMonitorFactory)Reflector.Invoke(obj, nameof(CreateServerMonitorFactory));
+
+        internal static Func<IStreamFactory, IStreamFactory> GetStreamFactoryWrapperFunc(this ClusterBuilder obj) => (Func<IStreamFactory, IStreamFactory>)Reflector.GetFieldValue(obj, "_streamFactoryWrapper");
```

Comment:
Go ahead and name this "_streamFactoryWrapper" following the naming in other Reflector classes.

In this case we prefer to name the the method after the the field name... even though method names don't normally start with an underscore. It's only a method because extension properties aren't supported yet.